Explicitly create KubeVirt service accounts token if not found

mfranczy opened the pull request:
**What this PR does / why we need it**:
K8s 1.24 disabled automatic token creation for service account,
thus it is required to create them by explicitly for CSI access

**Which issue(s) this PR fixes**:
<!--optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged-->
Fixes #10901 

**What type of PR is this?**
/kind bug
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:
<!--
Write your release note. Release notes are being used to generate the changelog:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Bugfix for KubeVirt infra CSI token creation due to auto-creation disabled in k8s 1.24.
```

**Documentation**:
<!--
Please do one of the following options:
- Add a link to the existing documentation
- Add a link to the kubermatic/docs pull request
- If no documentation change is applicable then add:
  - TBD (documentation will be added later)
  - NONE (no documentation needed for this PR)
-->
```documentation
NONE
```

mfranczy commented on pkg/provider/cloud/kubevirt/csi.go:
```diff
@@ -0,0 +1,163 @@
+const (
+	InfraStorageClassAnnotation = "kubevirt-initialization.k8c.io/initialize-sc"
+
+	csiServiceAccountNamespace = metav1.NamespaceDefault
```
This is another issue, CSI SA should be created in the cluster namespace. I will create follow-up PRs to refactor KubeVirt a little bit to change this.

hdurand0710 commented on pkg/provider/cloud/kubevirt/csi.go:
```diff
@@ -0,0 +1,163 @@
+package kubevirt
```
Small remark: we miss the header

hdurand0710 commented on pkg/provider/cloud/kubevirt/csi.go:
```diff
@@ -0,0 +1,163 @@
+	saCreators := []reconciling.NamedServiceAccountCreatorGetter{
+		csiServiceAccountCreator(csiResourceName),
+	}
+	if err := reconciling.ReconcileServiceAccounts(ctx, saCreators, csiServiceAccountNamespace, client); err != nil {
```
Do we take the opportunity of this PR to create the sa in the correct namespace ?

mfranczy replied:
I would prefer to not make this change in this PR. It will be a little bit bigger change.

hdurand0710 replied:
Fine for me.

hdurand0710 commented on pkg/provider/cloud/kubevirt/helper.go:
```diff
@@ -0,0 +1,40 @@
+package kubevirt
```
Same small remark for the missing header

hdurand0710 commented on pkg/provider/cloud/kubevirt/csi.go:
```diff
@@ -0,0 +1,163 @@
+package kubevirt
+
```
Can we merge this with `storage.go` ?

mfranczy replied:
storage.go doesn't exist anymore :)

hdurand0710 replied:
my bad then! 
But I don't see the removal of the file in the PR.
I see a removed `reconcile.go`, but not `storage.go`. How is it removed ?

mfranczy replied:
It's renamed to `data_volumes.go`

hdurand0710 replied:
oh wow, got it now, the way the diff appears is very misleading (at least to me  :-) ) perfect then.

mfranczy committed:
Explicitly create KubeVirt service accounts token if not found

K8s 1.24 disabled automatic token creation for service account,
thus it is required to create them by explicitly for CSI access

Signed-off-by: Marcin Franczyk <marcin0franczyk@gmail.com>
